lib/resin.coffee

```diff
@@ -57,17 +57,20 @@ sdkTemplate =
 	###
 	settings: require('./settings')
 
-module.exports = getSdk = (opts) ->
+module.exports = getSdk = (opts = {}) ->
 	defaults opts,
+		apiUrl: 'https://api.resin.io/'
 		apiVersion: 'v2',
-		isBrowser: window?
+		isBrowser: window?,
 
 	if opts.isBrowser
 		settings =
 			get: notImplemented
 			getAll: notImplemented
 	else
 		settings = require('resin-settings-client')
+		defaults opts,
+			settings.getAll()
```

Comment:
One reservation I have is that here we're lazy and _hope_ that the settings client uses the exact same names for the properties that are used by the SDK. While this assumption sounds reasonable it's not ensured by any contract.
I'd also rather use the explicit whitelist instead of blindly passing everything stored in the settings (that are general-purposed and can be used by the user's code).